receiver/telemetrygeneratorreceiver/config.go

```diff
@@ -206,81 +208,53 @@ func validateMetricsGeneratorConfig(g *GeneratorConfig) error {
 		return errors.New("metrics must be an array of maps")
 	}
 	for _, m := range metricsArray {
-		metric, ok := m.(map[string]any)
+		var metric metric
+		mapMetric, ok := m.(map[string]any)
 		if !ok {
 			return errors.New("each metric must be a map")
 		}
-		// check that the metric has a name
-		name, ok := metric["name"]
-		if !ok {
-			return errors.New("each metric must have a name")
-		}
-		// check that the metric has a type
-		metricType, ok := metric["type"]
-		if !ok {
-			return fmt.Errorf("metric %s missing type", name)
+
+		err := mapstructure.Decode(mapMetric, &metric)
+		if err != nil {
+			return fmt.Errorf("error decoding metric: %w", err)
 		}
-		// check that the metric type is valid
-		metricTypeStr, ok := metricType.(string)
-		if !ok {
-			return fmt.Errorf("metric %s has invalid metric type: %v", name, metricType)
+		if metric.Name == "" {
+			return errors.New("each metric must have a name")
 		}
-		switch metricTypeStr {
+
+		// validate the metric type
+		switch metric.Type {
 		case "Gauge", "Sum":
+		case "":
+			return fmt.Errorf("metric %s missing type", metric.Name)
 		default:
-			return fmt.Errorf("metric %s has invalid metric type: %s", name, metricTypeStr)
-		}
-		// check that the metric has a value_min
-		valMin, ok := metric["value_min"]
-		if !ok {
-			return fmt.Errorf("metric %s missing value_min", name)
+			return fmt.Errorf("metric %s has invalid metric type: %s", metric.Name, metric.Type)
 		}
-		// check that the value_min is a valid int
-		if _, ok = valMin.(int); !ok {
-			return fmt.Errorf("metric %s has invalid value_min: %v", name, valMin)
-		}
-		// check that the metric has a value_max
-		valMax, ok := metric["value_max"]
-		if !ok {
-			return fmt.Errorf("metric %s missing value_max", name)
-		}
-		// check that the value_max is a valid int
-		if _, ok = valMax.(int); !ok {
-			return fmt.Errorf("metric %s has invalid value_max: %v", name, valMax)
-		}
-		// check that the metric has a unit
-		unit, ok := metric["unit"]
-		if !ok {
-			return fmt.Errorf("metric %s missing unit", name)
-		}
-		// check that the unit is a valid string
-		unitStr, ok := unit.(string)
-		if !ok {
-			return fmt.Errorf("metric %s has invalid unit: %v", name, unit)
-		}
-		switch unitStr {
+
+		// validate the unit
+		switch metric.Unit {
 		case "By", "by", "1", "s", "{thread}", "{errors}", "{packets}", "{entries}", "{connections}", "{faults}", "{operations}", "{processes}":
+		case "":
+			return fmt.Errorf("metric %s missing unit", metric.Name)
 		default:
-			return fmt.Errorf("metric %s has invalid unit: %s", name, unitStr)
+			return fmt.Errorf("metric %s has invalid unit: %s", metric.Name, metric.Unit)
```

Comment:
I think you can have any metric you want in OTel as long as you put it in `{}`. If this is true we probably don't want this default case. Just a non-blank check.